test/rematch2.jl

```diff
@@ -197,22 +197,22 @@ end
             @test ex isa LoadError
             e = ex.error
             @test e isa ErrorException
-            @test e.msg == "$file:$line: Cannot infer which 2 of the 3 fields to match from any positional constructor for `$T207b`."
+            @test e.msg == "$file:$line: The type `$T207b` has 3 fields but the pattern expects 2 fields."
         end
     end
 end
 
-@testset "infer positional parameters from constructors 3" begin
+@testset "infer positional parameters from Rematch2.fieldnames(T) 3" begin
     # struct T207c
     #     x; y; z
     # end
     # T207c(x, y) = T207c(x, y, x)
     r = @match2 T207c(1, 2) begin
-        T207c(x, y) => x
+        T207c(x, y, z) => x
```

Comment:
it looks like the overload is `Rematch2.fieldnames(::Type{T207a}) = (:x, :y)`, so I'd expect this to not match `z` - what is the intention here?

Could we also add a case where the skipped field is in the middle of the struct's field list?

Reply:
This is `T207c`, not `T207a`.

I'll add one with a skip in the middle.